### PR TITLE
Added cumulocity-examples verification to pre-merge

### PR DESCRIPTION
**Pull request by kaczynskid-sag**
Also optimized pre-merge by improving parallelization

Causes https://github.com/SoftwareAG/cumulocity-examples/pull/86

### CODE REVIEW

File: .jenkins/github/pre-merge.jenkinsfile [hunk 1]
```diff
@@ -7,43 +7,96 @@ VERSIONS = [
...
+        resourceRequestMemory: '8Gi',
+        resourceLimitCpu: '4',
+        resourceLimitMemory: '8Gi',
+        workingDir: '/home/jenkins/agent'
+    ),
+    containerTemplate(
+        name: 'dind-daemon',
+        image: 'registry.stage.c8y.io/ci/docker:19.03-dind',
+        command: 'dockerd-entrypoint.sh',
+        args: '',
```
Comment (ghost) [inline .jenkins/github/pre-merge.jenkinsfile:39]: obsolete?

**Review by ghost — DISMISSED** [review]

File: .jenkins/github/pre-merge.jenkinsfile (hunk 1 above)
Comment (kaczynskid-sag) [reply .jenkins/github/pre-merge.jenkinsfile:39]: but harmless... I'd have to check the base image if it does not define any...

Comment (ghost) [reply .jenkins/github/pre-merge.jenkinsfile:39]: sure, don't bother

**Review by ghost — APPROVED** [review]

File: .jenkins/github/pre-merge.jenkinsfile [hunk 2]
```diff
@@ -210,27 +281,28 @@ def build_agents() {
...
 def spinup_instance(instance) {
-  if (!JOB_ABORTED && currentBuild.result != 'UNSTABLE' && currentBuild.result != 'FAILURE') {
-    stage("deploy $instance.name env") {
+  stage("spin up ${instance.label} instance") {
+    if (!JOB_ABORTED && currentBuild.result != 'UNSTABLE' && currentBuild.result != 'FAILURE') {
       build job: 'dev-x-cc-deployment', parameters: [
           string(name: 'DEPLOY_ENVIRONMENT', value: instance.name),
           string(name: 'VERSION', value: VERSIONS.release),
           string(name: 'BUILD_TYPE', value: 'custom'),
+          string(name: 'LP_LIST', value: ''),
```
Comment (d-mariak) [inline .jenkins/github/pre-merge.jenkinsfile:304]: Do we need that, if LP_LIST by default is empty?

**Review by d-mariak — APPROVED** [review]

File: .jenkins/github/pre-merge.jenkinsfile [hunk 3]
```diff
@@ -148,60 +181,98 @@ def build_sdk() {
...
+
+        dir('cumulocity-examples') {
+          stage('examples verify') {
+            try {
+              sh ".jenkins/scripts/mvn.sh install"
             } catch (e) {
-              stopPipeline('Publish agent artifacts', e)
+              unstable('Examples verification failed')
+            } finally {
+              junit(allowEmptyResults: true, checksName: "cumulocity-agents-unit-tests", keepLongStdio: true, testResults: '**/TEST-*.xml')
```
Comment (c8y-kh4mell) [inline .jenkins/github/pre-merge.jenkinsfile:275]: agents or examples ?

Comment (kaczynskid-sag) [reply .jenkins/github/pre-merge.jenkinsfile:275]: examples, missed a spot

**Comment by kaczynskid-sag** [comment]
I'll create a follow up PR to address those small issues, as I'd like to rebase the CVE related PR's on top of this change today to include the examples repo in verification of those changes as well.

**Pull request merged**